pkg/deploy/registry/deploylog/rest.go

```diff
@@ -180,6 +186,31 @@ func (r *REST) Get(ctx kapi.Context, name string, opts runtime.Object) (runtime.
 	}, nil
 }
 
+// waitForExistingDeployment will use the timeout to wait for a deployment to appear.
+func (r *REST) waitForExistingDeployment(namespace, name string) (*kapi.ReplicationController, error) {
+	var (
+		target *kapi.ReplicationController
+		err    error
+	)
+
+	condition := func() (bool, error) {
```

Comment:
Depends.  Think about from the user perspective.  Most of the time they want to see "the latest".  Failures are usually just us being too aggressive.  We don't have to retry everything, but a "if it errors, could we logically continue towards returning logs anyway" mindset is useful.

For example, in your scenario once we start returning logs we've satisfied the contract.  But if the logs call failed, it might be reasonable to start from the top again at least once to try to get logs.